src/main/java/org/myrobotlab/service/LocalSpeech.java

```diff
@@ -49,78 +46,197 @@
   private static final long serialVersionUID = 1L;
 
   public final static Logger log = LoggerFactory.getLogger(LocalSpeech.class);
-  private String ttsPath = System.getProperty("user.dir") + File.separator + "tts" + File.separator + "tts.exe";
+  protected String ttsPath = getResourceDir() + fs + "tts" + fs + "tts.exe";
+  protected String mimicPath = getResourceDir() + fs + "mimic" + fs + "mimic.exe";
   protected String ttsCommand = null;
+  protected String filterChars = "\"\'";
+  protected boolean removeExt = false;
+  protected boolean ttsHack = false;
 
   public LocalSpeech(String n, String id) {
     super(n, id);
+    // setup the default tts per os

```

Comment:
please avoid code in the constructor.. this is pretty light weight, but i feel it's a bad practice.  stuff like this should be in the startService method instead, not in the constructor